api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CXCASCRNCohortQueries.java

```diff
@@ -77,6 +78,392 @@ private CohortDefinition getA() {
     return cd;
   }
 
+  /**
+   * <b>SCRN_FR4</b>Patient with a screening test for Cervical cancer
+   *
+   * <ul>
+   *   <li>Have Resultado VIA with Positivo, Negativo or Suspeita de Cancro result marked on Ficha
+   *       de Registo Individual: Rastreio dos Cancros do Colo do Útero e da Mama during the
+   *       reporting period or
+   *   <li>Have Resultado do Rastreio HPV-DNA Negativo marked on Ficha de Registo Individual:
+   *       Rastreio dos Cancros do Colo do Útero e da Mama during the reporting period.
+   * </ul>
+   *
+   * <p><b>Note:</b>Patients with registration of tTreatment during the reporting period, but
+   * without VIA Result during the reporting period will not be included in CXCA_SCRN numerator.
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getPatientsWithScreeningTestForCervicalCancer(
+      CXCASCRNCohortQueries.CXCASCRNResult cxcascrnResult, boolean beforeStartDate) {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Patients with Screening Result for Cervical Cancer");
+    cd.addParameter(new Parameter("startDate", "startDate", Date.class));
+    cd.addParameter(new Parameter("endDate", "endDate", Date.class));
+    cd.addParameter(new Parameter("location", "location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("28", hivMetadata.getRastreioDoCancroDoColoUterinoEncounterType().getEncounterTypeId());
+    map.put("2094", hivMetadata.getResultadoViaConcept().getConceptId());
+    map.put("664", hivMetadata.getNegative().getConceptId());
+    map.put("703", hivMetadata.getPositive().getConceptId());
+    map.put("2093", hivMetadata.getSuspectedCancerConcept().getConceptId());
+    map.put("165436", hivMetadata.getHumanPapillomavirusDna().getConceptId());
+
+    StringBuilder query =
+        new StringBuilder(
+            "SELECT final.patient_id "
+                + "FROM   ( "
+                + "           SELECT   via_or_hpv.patient_id, "
+                + "                    Max(via_or_hpv.screening_date) AS last_screening_date "
```

Comment:
The requirements are not looking for the last screening date. Why using MAX function?

Reply:
Done